src/components/MySquads.js

```diff
@@ -21,26 +22,29 @@ const assignColor = () => {
 const MySquads = ({ userID }) => {
   const [userSquads, setUserSquads] = useState([]);
 
-  useEffect(() => {
-    console.log(userID);
-    getUserSquad(userID)
-      .then(({ data }) => {
-        const squads = data.map((attribute) => {
-          return {
-            competitive: attribute.attributes.squad.competitive
-              ? "Competitive"
-              : "Casual",
-            eventTime: attribute.attributes.squad["event_time"],
-            game: attribute.attributes.squad.game,
-            members: attribute.attributes.squad.members,
-            numberPlayers: attribute.attributes.squad["number_players"],
-          };
-        });
+  // React Native components don't unmount causing this component not to update then it's navigated to and from,
+  // useFocusEffect tells the component to do stuff when the user navigates to of from the screen
+  useFocusEffect(
+    React.useCallback(() => {
+      getUserSquad(userID)
+        .then(({ data }) => {
+          const squads = data.map((attribute) => {
+            return {
+              competitive: attribute.attributes.squad.competitive
+                ? "Competitive"
+                : "Casual",
+              eventTime: attribute.attributes.squad["event_time"],
+              game: attribute.attributes.squad.game,
+              members: attribute.attributes.squad.members,
+              numberPlayers: attribute.attributes.squad["number_players"],
+            };
+          });
 
-        setUserSquads(squads);
-      })
-      .catch((error) => console.log(error));
-  }, []);
+          setUserSquads(squads);
```

Comment:
@juliet-e Here is where the error / warning is occurring. Tom and I have put unique keys on all of the arrays in the `FlatList`s in this component. I also tried to just add a `key` key value pair to this object. I don't know why that would have worked, but I thought I would try. 
![IMG_0578](https://user-images.githubusercontent.com/95593550/197234057-ade96da4-5e78-43b3-8fb7-cd9d1fe74167.PNG)

Reply:
So I think I have narrowed it down to coming from the nested `FlatList` in the return. It finally occurred to me to comment that out. I believe that would be in the `MySquads` component between line 59-74 (ish).

Reply:
Ok I think I got it, we where using values that we thought where unique but they where now. I am going to push an addition into this PR that has changes to both how the keys are being created in the flatList and how the keys are being made.

I switched to using the `keyExtractor` prop, for example:
```js
keyExtractor={(memberData) =>
                  (memberData.gamertag + Math.random() * 100)
                }
```

I think this is the preferred method for add a key to a flat list array element, according to the React Native docs.

Reply:
Nice use of the keyExtractor and great job figuring that out!

Reply:
Nice job tracking down this bug and squashing it!

Reply:
Awesome job digging into this! If this consistently removes the issue - great. It seems a little weird to me that the key will not be static (Math.random() will evaluate to a different number every time) but maybe that won't be an issue. If you extracted the index in the flatList area, that would be unique, right? This is a small thing so if it works I wouldn't worry about it, but if any weirdness arises around having dynamically created keys with the Math.random, you could try just appending the gamertag with its index instead (`(memberData, index) => (memberData.gamertag + index)`. It's been a while since I've worked with React Native though so you can consider this comment a nit :)

Reply:
I agree it does feel weird. if we use the `gametag`, plus `index` there are cases where some of the gamer "avatar" views have the same key. I should have kept better documentation, but I think that also threw a warning. Since I am not 💯 on that though it is worth another shot.